lib/spack/spack/util/environment.py

```diff
@@ -461,12 +486,50 @@ def is_unset(self, var_name):
         # unset
         return (type(var_updates[-1]) == UnsetEnv)
 
+    def blacklist(self, banned):
+        """
+        Remove all modifications to variable names in banned.
+        """
+        self.env_modifications = list(filter(
+            lambda x: x.name not in banned, self.env_modifications))
+
     def clear(self):
         """
         Clears the current list of modifications
         """
         self.env_modifications = []
 
+    def reversed(self):
+        """
+        Returns the EnvironmentModifications object that will reverse self
+
+        Only creates reversals for additions to the environment, as reversing
+        ``unset`` and ``remove_path`` modifications is impossible.
+
+        Reversable operations are set(), prepend_path(), append_path(),
+        set_path(), and append_flags().
+        """
+        rev = EnvironmentModifications()
+
+        for envmod in reversed(self.env_modifications):
+            if type(envmod) == SetEnv:
+                rev.unset(envmod.name)
+            elif type(envmod) == AppendPath:
+                rev.remove_path(envmod.name, envmod.value)
+            elif type(envmod) == PrependPath:
+                rev.remove_path(envmod.name, envmod.value)
+            elif type(envmod) == SetPath:
+                for path in envmod.value:
```

Comment:
Is `SetPath.value` always an iterable or can it be a string? If it can be a string, this might fail

Reply:
It has to be an iterable